lib/msal-node/src/client/ClientApplication.ts

```diff
@@ -499,8 +503,12 @@ export abstract class ClientApplication {
 
         authRequest.authenticationScheme = AuthenticationScheme.BEARER;
 
-        // Set requested claims hash if claims were requested
-        if (authRequest.claims && !StringUtils.isEmpty(authRequest.claims)) {
+        // // Set requested claims hash if claims-based caching is enabled and claims were requested
+        if (
+            this.config.cache.claimsBasedCachingEnabled &&
+            authRequest.claims &&
+            !StringUtils.isEmpty(authRequest.claims)
```

Comment:
```suggestion
            authRequest.claims
```

Reply:
Changing to check for empty stringified object instead, which I think was the original intention that I misinterpreted back when I first added claims-based caching